File: AiSD5_C2/exercise_3/Pascal/main.pas
program Suma_cyfr_liczby;

function CzytajLiczbe: integer;
var
  Liczba: integer;
begin
    writeln;
    writeln('Oblicznaie sumy cyfr liczby calkowitej');
    write('Podaj liczbe:'); readln(Liczba);
    CzytajLiczbe:=Liczba;
end;

function SumaCyfr(Liczba: integer): integer;
var
  Suma: integer;
begin
    Suma := 0;
    repeat
        Suma := Suma + ( Liczba mod 10 );
        Liczba := Liczba div 10;
    until Liczba = 0;
    SumaCyfr := Suma;
end;

procedure PiszSume(Liczba, SumaCyfr: integer);
begin
    writeln('Suma cyfr liczby ', Liczba, ' wynosi ', SumaCyfr);
end;

procedure SumaCyfrLiczby;
var
  liczba, suma: integer;
begin
    Liczba:=CzytajLiczbe;
    Suma :=SumaCyfr(liczba);
    PiszSume(Liczba, Suma)
end;

begin
SumaCyfrLiczby;
readln;
end.

File: AiSD5_C2/exercise_3/Python/main.py
def czytaj_liczbe():
    print('\nObliczanie sumy cyfr liczby calkowitej')
    liczba = int(input('Podaj liczbe: '))
    return liczba

def suma_cyfr(liczba):
    suma = 0
    liczba = abs(liczba)  # Użyj wartości bezwzględnej
    while liczba != 0:
        suma += liczba % 10
        liczba //= 10
    return suma

def pisz_sume(liczba, suma_cyfr):
    print(f'Suma cyfr liczby {liczba} wynosi {suma_cyfr}')

def suma_cyfr_liczby():
    liczba = czytaj_liczbe()
    suma = suma_cyfr(liczba)
    pisz_sume(liczba, suma)

if __name__ == "__main__":
    suma_cyfr_liczby()